### PR TITLE
Exploit reporting

### PR DESCRIPTION
**Pull request by Selora**
Added reporting capabilities.
Capturing MSF output and putting it in "report.csv"
Also create an MSF RCscript for every modules ran against a given host. That way, you can reproduce whatever caused an exploit to work.

### CODE REVIEW

**Comment by Ekultek** [comment]
This line as well https://github.com/NullArray/AutoSploit/pull/116/files#diff-d518d8fdb7b9cda0b426f1945c00e069R10

**Comment by Selora** [comment]
It should be compatible with Python3.x, now it's a bit harder to test since raw_inputs is used elsewhere in the original codebase.

I haven't made use of python2-specific features so it shouldn't be a problem to port.

Regarding the modification of import statements, can you please elaborate on what needs to be done? I've read this chapter and I'm not seeing what needs to be modified.
https://www.python.org/dev/peps/pep-0008/#imports

**Comment by Ekultek** [comment]
Here’s our code standards https://github.com/NullArray/AutoSploit/wiki/Development-information#contribution-standards

**Comment by Selora** [comment]
I modified the two import lines as asked. It's the same commit ID, I forced-pushed it after an amended commit.

**Comment by Ekultek** [comment]
I'll get it tested soon, gotta install a VM first

**Comment by Ekultek** [comment]
Did you originally fork from master?

**Comment by Ekultek** [comment]
I like it put there's a couple things I don't like about it.

File: lib/settings.py [hunk 1]
```diff
@@ -166,14 +171,19 @@ def cmdline(command):
...
-    os.system(command)
-    '''process = subprocess.call(
-        args=" ".join(command),
-        stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT,
-        shell=True
-    )
-    return process'''
+    #os.system(command)
+    lib.output.info("Executing command {}".format(command))
```
Comment (Ekultek) [inline lib/settings.py]: This command needs to be wrapped in single quotes and stripped example:

`executing command 'sudo msfconsole <SOMETHING> <OPTS>'`

File: lib/settings.py [hunk 2]
```diff
@@ -166,14 +171,19 @@ def cmdline(command):
...
+    split_cmd = [x.strip() for x in command.split(" ") if x]
+
+    sys.stdout.flush()
+
+    proc = Popen(split_cmd, stdout=PIPE, bufsize=1)
+    stdout_buff = []
+    for stdout_line in iter(proc.stdout.readline, b''):
+        stdout_buff += [stdout_line.rstrip()]
+        print(">>>> {}".format(stdout_line).rstrip())
```
Comment (Ekultek) [inline lib/settings.py]: We need an output image for this, something like:

`(msf)>> OUTPUT`

**Comment by Ekultek** [comment]
Decided I'd just do it myself, lemme retest real quick

**Comment by Ekultek** [comment]
One more thing, this is going to cause a whole shit load of directories to be created in `autosploit_out` would leaving off the hour minute and second fuck with anything on it? Or would it just create new directories for each IP address in there?

**Comment by Ekultek** [comment]
I'm gonna go ahead and merge this, I'll fix stuff I don't like later on.

**Comment by Ekultek** [comment]
Yeah I saw it, good job I like it.

**Pull request merged**